wrappers/android/examples/devices_presets/src/main/java/com/intel/realsense/devices_presets/MainActivity.java

```diff
@@ -0,0 +1,219 @@
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.content.res.Resources;
+import android.os.Bundle;
+import android.text.Html;
+import android.util.Log;
+import android.view.View;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+import com.intel.realsense.librealsense.Device;
+import com.intel.realsense.librealsense.Option;
+import com.intel.realsense.librealsense.RsContext;
+import com.intel.realsense.librealsense.Sensor;
+import com.intel.realsense.librealsense.DeviceList;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+
+public class MainActivity extends AppCompatActivity {
+
+    private static final String TAG = "librs presets for devices example";
+    private static final int PERMISSIONS_REQUEST_CAMERA = 0;
+
+    private boolean mPermissionsGranted = false;
+
+
+    private static int mSelectedItem = 500;
+    private RsContext mRsContext;
+    private TextView deviceResponseET;
+    private String[] mPresets;
+    private Resources mResources;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        // Android 9 also requires camera permissions
+        if (android.os.Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.O &&
+                ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.CAMERA}, PERMISSIONS_REQUEST_CAMERA);
+            return;
+        }
+        mPermissionsGranted = true;
+
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.CAMERA}, PERMISSIONS_REQUEST_CAMERA);
+            return;
+        }
+        mPermissionsGranted = true;
+        init();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if(mPermissionsGranted)
+            init();
+        else
+            Log.e(TAG, "missing permissions");
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+    }
+
+    private void init(){
+        RsContext.init(this);
+        mRsContext = new RsContext();
+
+        TextView message = findViewById(R.id.presets_list_title);
+        deviceResponseET = findViewById(R.id.deviceResponse);
+        mResources = getResources();
+        mPresets = null;
+        mPresets = new String[]{};
```

Comment:
It seems that the assignment to null is not necessary before assignment with new